### DISCUSSION TITLE
v2.17

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#1227] [General]
# 2.17

# Big changes

* Indicate status of VBus for modding Pinecil (debug menu)
* Better hall effect sensor sensitivity adjustment (larger range with more steps)
* Temperature increment will "round" to nearest multiple of increase amount
* Build setup migrated to Alpine (You can now build in docker easily, and on PinePhone/PinePhonePro)
* -> Removed proprietary compiler for Pinecil RISCV now all uses normal gcc
* -> Removed using the arm specific build of gcc for the one that alpine ships (Miniware devices)
* Logo generator python script creates `.dfu` files for ease of use with Pinecil
* Upgrades to translations
* Support for new GD32103 based TS100 units turning up on the market
* Raw hall effect reading now shows in the Pinecil debug menu
* Fixed automatic orientation for newer TS80P's with the SC7 accelerometer
* User interface slight changes
* New `metadata.zip` file to allow the Pine Updater to automatically fetch information on releases

## Notes

* VBus mod detection may not play well with all PPS chargers. If your iron reboots when you view this in the debug menu its not a _fault_. (#1226)
* `metadata.zip` is only designed for use by automatic software, ignore it for normal use
* More details on Pinecil VBus mod coming via other channels.
* Hall effect sensor is not fitted to Pinecil's by default, you have to fit this yourself if you want the feature
* Tweaks to the Accelerometer code means the drivers are slightly more fussy. If you run into any issues let us know in the discussion or issues.

## Full changelog

* bug_report.md: minor cleanup by @evils in https://github.com/Ralim/IronOS/pull/1161
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/1163
* fixed detailed view for lefties by @discip in https://github.com/Ralim/IronOS/pull/1165
* Update translation_DE.json by @discip in https://github.com/Ralim/IronOS/pull/1166
* Fix flash ts100 linux by @ysard in https://github.com/Ralim/IronOS/pull/1167
* Settings by @Ralim in https://github.com/Ralim/IronOS/pull/1164
* Changed default orientation to Auto for TS100, TS80(P) & Pinecil ... by @discip in https://github.com/Ralim/IronOS/pull/1168
* changed the source of the DisconnectedTip Icon to reduce code 😊 by @discip in https://github.com/Ralim/IronOS/pull/1171
* Migrating to Alpine build env by @Ralim in https://github.com/Ralim/IronOS/pull/1169
* updated HomeScreen.png by @discip in https://github.com/Ralim/IronOS/pull/1172
* updated and added logos by @discip in https://github.com/Ralim/IronOS/pull/1174
* updated some entries by @discip in https://github.com/Ralim/IronOS/pull/1175
* Update Logo.md - simplify the objcopy command by @HoroTW in https://github.com/Ralim/IronOS/pull/1176
* Add capability of writing logos in binary format by @cybernesto in https://github.com/Ralim/IronOS/pull/1178
* Patch settings decrement to honour increment size and not skip max by @Ralim in https://github.com/Ralim/IronOS/pull/1181
* added preview for the logos & an animated logo + hex-files ... by @discip in https://github.com/Ralim/IronOS/pull/1180
* Hungarian translateion update v2 by @hippi-viking in https://github.com/Ralim/IronOS/pull/1185
* Update Flashing.md by @discip in https://github.com/Ralim/IronOS/pull/1189
* idlescreenBG  -> buttonA & buttonB by @discip in https://github.com/Ralim/IronOS/pull/1188
* Updating translation editor by @Ralim in https://github.com/Ralim/IronOS/pull/1183
* Update translation_RU.json by @JugglerLKR in https://github.com/Ralim/IronOS/pull/1197
* Convert logos directly into DFU format by @cybernesto in https://github.com/Ralim/IronOS/pull/1190
* TS100 GD32 by @Ralim in https://github.com/Ralim/IronOS/pull/1202
* fixed a typo by @discip in https://github.com/Ralim/IronOS/pull/1204
* Add raw hall effect output to the debug menu. by @ritkit in https://github.com/Ralim/IronOS/pull/1196
* HALL only on Pinecil by @discip in https://github.com/Ralim/IronOS/pull/1205
* TS100 warning fix (#1192) by @Ralim in https://github.com/Ralim/IronOS/pull/1206
* Cleanup OLED_FLIP logic by @Ralim in https://github.com/Ralim/IronOS/pull/1207
* Fix lefty mode again by @Ralim in https://github.com/Ralim/IronOS/pull/1208
* Format to alpine by @Ralim in https://github.com/Ralim/IronOS/pull/1211
* Pinecil mod detection by @Ralim in https://github.com/Ralim/IronOS/pull/1215
* Create json meta by @Ralim in https://github.com/Ralim/IronOS/pull/1218
* Merge json metadata into one file by @Ralim in https://github.com/Ralim/IronOS/pull/1222
* Enable SC7 for Miniware by @Ralim in https://github.com/Ralim/IronOS/pull/1219
* Update Hall sensor tuning by @Ralim in https://github.com/Ralim/IronOS/pull/1216
* logo update by @discip in https://github.com/Ralim/IronOS/pull/1223
* Only vbus mod check on pinecil by @Ralim in https://github.com/Ralim/IronOS/pull/1224
* TS80P Accelerometer version detection again by @Ralim in https://github.com/Ralim/IronOS/pull/1225

## New Contributors

* @evils made their first contribution in https://github.com/Ralim/IronOS/pull/1161
* @ysard made their first contribution in https://github.com/Ralim/IronOS/pull/1167
* @HoroTW made their first contribution in https://github.com/Ralim/IronOS/pull/1176
* @cybernesto made their first contribution in https://github.com/Ralim/IronOS/pull/1178
* @JugglerLKR made their first contribution in https://github.com/Ralim/IronOS/pull/1197
* @ritkit made their first contribution in https://github.com/Ralim/IronOS/pull/1196

**Full Changelog**: https://github.com/Ralim/IronOS/compare/v2.16...v2.17

Special thanks to goes out to everyone who contributed their time or financial support. The community makes this project what it is ❤️ 

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.17'>v2.17 </a>.</em>

**Comment by ruslan61rus** [comment]
Today i tried to flash pinecil, with dfu tool, but unsuccessfully... Anything changes in flashing method? Before, with 2.16 release, flashing was succesfully.

**Reply by Ralim** [reply]
Should be no changes from 2.16.
What errors are you getting ?

**Comment by ruslan61rus** [comment]
Got it. Do not need tick "erase all", and then it flashes successfully)

**Comment by Eugene-Market** [comment]
IT'S ALIVE HAHAHAHAHAHA THANK YOU

**Comment by Aleksandr2104** [comment]
Problem with transferring the firmware file to TS100. Tried on different computers on Win10 and Win7. Writes that there is no way to the soldering iron

**Reply by Ralim** [reply]
Hello,
Most likely you are running into one of the numerous issues with the Miniware bootloader.

Please search the existing issues on this GitHub repository, it has been discussed many times and some of the comments by others should help.

**Reply by JugglerLKR** [reply]
> Problem with transferring the firmware file to TS100. Tried on different computers on Win10 and Win7. Writes that there is no way to the soldering iron

Ubuntu 18 is the solution.

**Comment by logser13** [comment]
For ts80p russian version 2.17 does not work the motion sensor, but in version 2.16 (with Russian) works.

**Reply by Ralim** [reply]
Are you affected by this : https://github.com/Ralim/IronOS/issues/1229

**Comment by deanZZZZZ** [comment]
Tilt angle sensor not work  on MHP30

**Reply by Ralim** [reply]
Hello, have you tested the latest release (2.18?)

**Reply by gordon-quad** [reply]
Does not work for me with 2.18.70565DE (18-04-2022) downloaded from GitHub Releases.

**Reply by Ralim** [reply]
Hia,
I think this is fixed now, where tilting will cause unit to shutdown.
Can you give https://github.com/Ralim/IronOS/actions/runs/2454469240 a try

**Comment by deanZZZZZ** [comment]
yes..same 2.18 writes that it does not detect the sensor
